src/DotNetEnv/IEnvReader.cs

```diff
@@ -0,0 +1,200 @@
+        int GetIntValue(string variable);
+
+        /// <summary>
+        /// Gets the value of an environment variable in double format.
+        /// </summary>
+        /// <param name="variable">The variable of the value to get.</param>
+        /// <returns>The value of the variable in double format.</returns>
+        /// <exception cref="EnvVariableNotFoundException">variable is not found in the current process.</exception>
+        double GetDoubleValue(string variable);
+
+        /// <summary>
+        /// Try to retrieve the value of an environment variable in string format.
+        /// </summary>
+        /// <param name="variable">The variable of the value to try retrieve.</param>
+        /// <param name="value">The string value retrieved or null.</param>
+        /// <returns>true if the environment variable exists in the current process, otherwise false.</returns>
+        bool TryGetStringValue(string variable, out string value);
+
+        /// <summary>
+        /// Try to retrieve the value of an environment variable in boolean format.
+        /// </summary>
+        /// <param name="variable">The variable of the value to try retrieve.</param>
+        /// <param name="value">The boolean value retrieved or false.</param>
+        /// <returns>true if the environment variable exists in the current process, otherwise false.</returns>
+        bool TryGetBoolValue(string variable, out bool value);
+
+        /// <summary>
+        /// Try to retrieve the value of an environment variable in integer format.
+        /// </summary>
+        /// <param name="variable">The variable of the value to try retrieve.</param>
+        /// <param name="value">The integer value retrieved or zero.</param>
+        /// <returns>true if the environment variable exists in the current process, otherwise false.</returns>
+        bool TryGetIntValue(string variable, out int value);
+
+        /// <summary>
+        /// Try to retrieve the value of an environment variable in double format.
+        /// </summary>
+        /// <param name="variable">The variable of the value to try retrieve.</param>
+        /// <param name="value">The double value retrieved or zero.</param>
+        /// <returns>true if the environment variable exists in the current process, otherwise false.</returns>
+        bool TryGetDoubleValue(string variable, out double value);
+    }
+
+    /// <inheritdoc cref="IEnvReader" />
+    public class EnvReader : IEnvReader
+    {
+        private const string EnvVariableNotFoundMessage = "The value could not be retrieved because it does not exist in the current process.";
+        private static EnvReader instance;
+
+        /// <summary>
+        /// Gets an instance of type EnvReader.
+        /// </summary>
+        public static EnvReader Instance
+        {
+            get
+            {
+                if (instance == null)
```

Comment:
This is not thread safe. You should read up on how to make a singleton thread safe, if you really must have a singleton at all -- I strongly discourage it, personally.

Reply:
To avoid complications, I have removed the property.